app/models/generic_object_definition.rb

```diff
@@ -37,7 +37,7 @@ def properties=(props)
   end
 
   def type_cast(attr_name, value)
-    TYPE_MAP.fetch(defined_property_attributes[attr_name]).cast(value)
+    TYPE_MAP.fetch(defined_property_attributes[attr_name]).new.cast(value)
```

Comment:
Isn't the old implementation to instantiate the type objects in `TYPE_MAP` better for performance?